lib/models/ui/create_account_errors.dart

```diff
@@ -0,0 +1,40 @@
+import "package:flutter/foundation.dart";
+
+/// A dataclass handling the error status of the create account page.
+@immutable
+class CreateAccountErrors {
+  final String? uniqueUsernameError;
+  final String? pseudoError;
+  final bool valid;
```

Comment:
Good job ! 
I think it's a very good idea to have a separated class to represent the errors state :+1: 

I have just a small suggestion:
I believe that you are using the `valid` field to represent two different concepts:
- If the account fields are valid
- If the account has been successfully added to the database

I think that we could separate those two concepts by adding a field `created` (that is true when the account has been created on the database)
And also this way we may want to make the field `valid` a function as below:

```suggestion
  final bool created;
  bool get valid => uniqueUsernameError == null && pseudoError == null;
```

If you agree with this suggestion, it might be more explicit to rename the class `CreateAccountState`.

Reply:
This is indeed a very interesting idea. I realise that the `valid` variable does not have much purpose currently, and adding a `created` allows to reason more easily about the code.